simulator-starter/src/test/java/com/consol/citrus/simulator/service/MessageRepositoryTest.java

```diff
@@ -0,0 +1,191 @@
+
+        Assert.assertEquals(1, result.size());
+        Assert.assertEquals(uid, result.get(0).getCitrusMessageId());
+        Assert.assertEquals(PAYLOAD, result.get(0).getPayload());
+        
+        filter.setHeaderFilter(HEADER_NAME+":"+uid+";"+HEADER_NAME2+":"+uid+"_3");
+        
+        result = messageRepository.find(filterAdapter);
+        Assert.assertEquals(0, result.size());
+    }
+
+    @Test
+    void testFindByHeaderLike() {
+        String innerUid = UUID.randomUUID().toString();
+        String uid = createTestMessage("PRE" + innerUid + "POST", PAYLOAD);
+
+        MessageFilter filter = new MessageFilter();
+        filter.setDirectionOutbound(false);
+
+        filter.setHeaderFilter(HEADER_NAME+":"+"%" + innerUid + "%");
+        
+        MessageFilter filterAdapter = QUERY_FILTER_ADAPTER_FACTORY.getQueryAdapter(filter);
+        List<Message> result = messageRepository.find(filterAdapter);
+
+        Assert.assertEquals(1, result.size());
+        Assert.assertEquals(uid, result.get(0).getCitrusMessageId());
+        Assert.assertEquals(PAYLOAD, result.get(0).getPayload());
+    }
+
+    @Test
+    void testFindByPayloadLike() {
+        String uid = UUID.randomUUID().toString();
+        String specificPayload = "Pay" + uid + "LOAD";
+        uid = createTestMessage(uid, specificPayload);
+
+        MessageFilter filter = new MessageFilter();
+        filter.setDirectionOutbound(false);
+        filter.setContainingText("%" + uid + "%");
+        
+        MessageFilter filterAdapter = QUERY_FILTER_ADAPTER_FACTORY.getQueryAdapter(filter);
+        List<Message> result = messageRepository.find(filterAdapter);
+
+        Assert.assertEquals(1, result.size());
+        Assert.assertEquals(uid, result.get(0).getCitrusMessageId());
+        Assert.assertEquals(specificPayload, result.get(0).getPayload());
+    }
+
+    @Test
+    void testFindByPayload() {
+        String uid = createTestMessage();
+
+        MessageFilter filter = new MessageFilter();
+        filter.setDirectionOutbound(false);
+        filter.setContainingText(PAYLOAD);
+        
+        MessageFilter filterAdapter = QUERY_FILTER_ADAPTER_FACTORY.getQueryAdapter(filter);
+
+        List<Message> result = messageRepository.find(filterAdapter);
+
+        Assert.assertEquals(1, result.size());
+        Assert.assertEquals(uid, result.get(0).getCitrusMessageId());
+        Assert.assertEquals(PAYLOAD, result.get(0).getPayload());
+    }
+
+    @Test
+    void testFindByAllParams() {
+        Date startSavingDate = now();
+        String uid = createTestMessage();
+        Date endSavingDate = now();
+
+        // Filter by all valid
+        MessageFilter filter = new MessageFilter();
+        filter.setFromDate(startSavingDate);
+        filter.setToDate(endSavingDate);
+        filter.setDirectionOutbound(false);
+        filter.setContainingText(PAYLOAD);
+        filter.setHeaderFilter(HEADER_NAME+":"+uid);
```

Comment:
formatting.. I think you didn't format the entire file.

Reply:
done